src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -1089,19 +1037,67 @@ public void SetEditorCellVisibility(bool shown)
     /// <summary>
     ///   Changes the number of mutation points left. Should only be called by MicrobeEditorAction
     /// </summary>
-    internal void ChangeMutationPoints(int change)
+    internal void UpdateMutationPoints()
     {
         if (FreeBuilding || CheatManager.InfiniteMP)
             return;
 
-        MutationPoints = (MutationPoints + change).Clamp(0, Constants.BASE_MUTATION_POINTS);
+        MutationPoints = MutationPointsAfterChange(CurrentSpecies);
 
         gui.UpdateMutationPointsBar();
     }
 
+    internal int MutationPointsAfterChange(MicrobeSpecies newSpecies)
+    {
+        var rigidityChangeCost = (int)Math.Round(
+            Math.Abs(newSpecies.MembraneRigidity - startingSpecies.MembraneRigidity)
+            * Constants.MEMBRANE_RIGIDITY_COST_PER_STEP * Constants.MEMBRANE_RIGIDITY_SLIDER_TO_VALUE_RATIO);
+
+        var membraneChangeCost = startingSpecies.MembraneType.Equals(newSpecies.MembraneType) ?
+            0 : newSpecies.MembraneType.EditorCost;
+
+        // Calculate which organelles are moved, added, or deleted
+        var commonOrganelles = startingSpecies.Organelles.Organelles
+            .Where(x => newSpecies.Organelles.Organelles.Contains(x));
+
+        // Organelles that are not in the exact same spot; either deleted or moved
+        var missingOrganelles = startingSpecies.Organelles.Organelles.Where(x => !commonOrganelles.Contains(x));
+
+        List<OrganelleTemplate> movedOrganelles = new List<OrganelleTemplate>();
+        List<OrganelleTemplate> addedOrganelles = new List<OrganelleTemplate>();
+
+        // Search the new organelles for old ones of the same time, to count as many as having moved as possible
+        // ReSharper disable once PossibleMultipleEnumeration
```

Comment:
I don't think there's anything I can do about this one. JetBrains mentions false positives with this rule: https://www.jetbrains.com/help/rider/PossibleMultipleEnumeration.html#false-positives 

And I don't see anything being double enumerated here. With the fact that this calculation is only being done once per player input, I'm not concerned about performance here anyway.

Reply:
Multiple enumeration warning goes away if you strategically add a "ToList" call in the right place so that the enumeration is done once and stored in the list then.